test/apps/uct_info/uct_info.c

```diff
@@ -0,0 +1,199 @@
+/**
+* Copyright (C) 2022, NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <alloca.h>
+#include <string.h>
+
+#include <uct/api/uct.h>
+#include <ucs/sys/string.h>
+#include <ucs/sys/compiler_def.h>
+
+
+#define CALL(_action, _err) \
+    do { \
+        ucs_status_t UCS_V_UNUSED _status; \
+        _status = _action; \
+        if (_status != UCS_OK) { \
+            _err; \
+        } \
+    } while (0)
+
+
+#define UCT_CALL(_action, _err) \
+    CALL(_action, { \
+        printf("ERROR: %s failed\n", #_action); \
+        _err; \
+    })
+
+
+static ucs_status_t uct_info_iface_info(uct_worker_h worker, uct_md_h md,
+                                        uct_tl_resource_desc_t *resource)
+{
+    uct_iface_params_t iface_params = {
+        .field_mask           = UCT_IFACE_PARAM_FIELD_OPEN_MODE |
+                                UCT_IFACE_PARAM_FIELD_DEVICE,
+        .open_mode            = UCT_IFACE_OPEN_MODE_DEVICE,
+        .mode.device.tl_name  = resource->tl_name,
+        .mode.device.dev_name = resource->dev_name
+    };
+    ucs_status_t status             = UCS_OK;
+    uct_iface_config_t *iface_config;
+    uct_iface_h iface;
+
+    UCT_CALL(uct_md_iface_config_read(md, resource->tl_name, NULL, NULL,
+                                      &iface_config),
+             return _status);
+
+    printf("#      Transport: %s\n", resource->tl_name);
+    printf("#         Device: %s\n", resource->dev_name);
+    printf("#           Type: %s\n", uct_device_type_names[resource->dev_type]);
+    printf("#  System device: %s",
+           ucs_topo_sys_device_get_name(resource->sys_device));
+    printf("\n");
```

Comment:
why need to print it separately from the previous line print?

Reply:
> why need to print it separately from the previous line print?

removed